spacemesh/v2alpha1/smeshing_identities.proto

```diff
@@ -0,0 +1,50 @@
+syntax = "proto3";
+
+package spacemesh.v2alpha1;
+
+import "google/api/annotations.proto";
+import "google/api/visibility.proto";
+import "google/protobuf/timestamp.proto";
+
+message IdentityStatesRequest {}
+
+enum IdentityState {
+  UNSPECIFIED = 0;
+
+  WAIT_FOR_ATX_SYNCED = 1;
+  RETRYING = 2;
+
+  WAITING_FOR_POET_REGISTRATION_WINDOW = 3;
+  POET_CHALLENGE_READY = 4;
+  POET_REGISTERED = 5;
+  WAIT_FOR_POET_ROUND_END = 6;
+  POET_PROOF_RECEIVED = 7;
+
+  GENERATING_POST_PROOF = 8;
+  POST_PROOF_READY = 9;
+
+  ATX_READY = 10;
+  ATX_BROADCASTED = 11;
+}
+
+message IdentityStateInfo {
+  IdentityState state = 1;
+  optional uint32 publish_epoch = 2;
+  string message = 3;
+  google.protobuf.Timestamp time = 4;
+}
+
+message Identity {
+  bytes smesher_id = 1;
+  repeated IdentityStateInfo history = 2;
+}
+
+message IdentityStatesResponse {
+  repeated Identity identities = 3;
+}
```

Comment:
How about using a [map](https://protobuf.dev/programming-guides/proto3/#maps)?
```suggestion
message Identity {
  repeated IdentityStateInfo history = 1;
}

message IdentityStatesResponse {
  map<string, Identity> identities = 1;
}
```

The downside is that a map key cannot be `bytes` so you would need to encode it (e.g in hex). But on the other side, it would safe space because you wouldn't need to repeat the smesher_id.